src/main/java/org/reactome/release/downloaddirectory/GenerateGOAnnotationFile/GOAGeneratorUtilities.java

```diff
@@ -186,46 +245,51 @@ public static Integer assignDateForGOALine(GKInstance entityInst, String goaLine
      * @return -- Integer, from instanceEdit's dateTime. Parsed to remove the Timestamp.
      * @throws Exception -- MySQLAdaptor exception.
      */
     private static int getDate(GKInstance instanceEditInst) throws Exception {
-        return Integer.valueOf(instanceEditInst.getAttributeValue(ReactomeJavaConstants.dateTime).toString().split(" ")[0].replaceAll("-", ""));
+        return Integer.valueOf(
+                instanceEditInst.getAttributeValue(ReactomeJavaConstants.dateTime)
+                        .toString()
+                        .split(" ")[0]
+                        .replaceAll("-", "")
+        );
     }
 
-    /**
-     * Iterates through the lines in the 'goaLines' list, retrieves the date associated with that line and also adds the 'Reactome' column before adding it to the gene_association.reactome file.
-     * @throws IOException -- File writing/reading exceptions.
-     */
-    public static void outputGOAFile() throws IOException {
-
-        Path goaFilepath = Paths.get(GOA_FILENAME);
-
-        final String GAF_HEADER = "!gaf-version: 2.1\n";
+    private static void writeHeader(Path goaFilepath) throws IOException {
+        final String gafHeader = String.join(System.lineSeparator(),
+            "!gaf-version: 2.2",
+            "generated-by: Reactome",
+            "date-generated: " + getCurrentDateAsYYYYMMDD()
+        ).concat(System.lineSeparator());
 
-        Files.deleteIfExists(goaFilepath);
-        Files.write(goaFilepath, GAF_HEADER.getBytes());
-
-        List<String> lines = goaLines.stream().sorted().map(
-                line -> String.join("\t", line, dates.get(line).toString(), REACTOME_STRING, "","")
-        ).collect(Collectors.toList());
-        Files.write(goaFilepath, lines, StandardOpenOption.APPEND);
+        Files.write(goaFilepath, gafHeader.getBytes(), StandardOpenOption.APPEND);
+    }
 
-        gzipGOAFile();
+    private static String getCurrentDateAsYYYYMMDD() {
+        return Instant.now().atZone(ZoneId.systemDefault()).format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
     }
 
     /**
      * Gzips gene_association.reactome file
+     * @param goaFilePath Path to the unzipped gene_association.reactome file
      * @throws IOException -- File writing/reading exceptions.
      */
-    private static void gzipGOAFile() throws IOException {
-        File goaFile = new File(GOA_FILENAME);
-        File goaFileGZipped = new File(GOA_FILENAME + ".gz");
-        GZIPOutputStream gzipOutputStream = new GZIPOutputStream(new FileOutputStream(goaFileGZipped));
-        try (FileInputStream fileInputStream = new FileInputStream(goaFile)) {
+    private static void gzipGOAFile(Path goaFilePath) throws IOException {
+        Path goaFilePathGZipped = Paths.get(goaFilePath.toAbsolutePath().toString() + ".gz");
+        GZIPOutputStream gzipOutputStream = new GZIPOutputStream(new FileOutputStream(goaFilePathGZipped.toFile()));
```

Comment:
A try-with-resources block could be used here, and then you wouldn't need to explicitly close gzipOutputStream.

Reply:
Done!